### PR TITLE
Made google-services.json as  optional file.

### LINKED ISSUE #43: Make adding google-services.json optional
Currently it is mandatory to add google-services to the project. We should make it optional.

### PR DESCRIPTION
**Pull request by 098suraj**
Fixes #43 

#### What has been done to verify that this works as intended?
       I have built and tested the sample app for the basic functionalities and it seems to be working for all the current features. I 
      have also manually verified that the optional Google services JSON file is handled correctly and does not cause any issues or 
      crashes.
#### Why is this the best possible solution? Were any other approaches considered?
         For making the Google services JSON file optional, we need to check if it exists, and based on that, we apply plugins and 
        dependencies for the same. This is the best possible solution as it provides flexibility for developers to include or exclude 
        the file as needed, without causing any issues with the app.
#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?
         This change makes the Google services JSON file optional, which means that developers can choose to include or exclude 
        it as needed. This should not affect the behavior of the app in any significant way. However, there is a risk of regression if 
       the code changes inadvertently affect the handling of the optional file or if there are issues with the dependencies that are 
       included/excluded based on the presence of the file.

### CODE REVIEW

File: sample/build.gradle [hunk 1]
```diff
@@ -23,7 +23,13 @@ if (secretsFile.exists()) {
 }
 def googleMapsApiKey = secrets.getProperty('GOOGLE_MAPS_API_KEY', '')
 def mapboxAccessToken = secrets.getProperty('MAPBOX_ACCESS_TOKEN', '')
-
+def googleServicesFile = rootProject.file('google-services.json')
+if (googleServicesFile.exists()) {
+    apply plugin: 'com.google.gms.google-services'
+    apply plugin: 'com.google.firebase.crashlytics'
+} else {
+    println("google-services.json not found .Firebase services will not be availabe ")
```
Comment (chinmoy12c) [inline sample/build.gradle]: nit, please correct the position of fullstop.

File: sample/build.gradle [hunk 2]
```diff
@@ -190,7 +198,7 @@ dependencies {
     implementation Dependencies.bikram_sambat
     implementation Dependencies.rarepebble_colorpicker
     implementation Dependencies.commons_io
-    implementation (Dependencies.opencsv) {
```
Comment (chinmoy12c) [inline sample/build.gradle]: nit add the deleted space.

File: sample/build.gradle [hunk 3]
```diff
@@ -287,5 +295,3 @@ dependencies {
 //    debugImplementation Dependencies.leakcanary
 }
 
-apply plugin: 'com.google.gms.google-services'
-apply plugin: 'com.google.firebase.crashlytics'
```
Comment (chinmoy12c) [inline sample/build.gradle:291]: nit add a newline at the end of file.

**Review by chinmoy12c — CHANGES_REQUESTED** [review]
Hey @098suraj , thanks for the PR. Currently, if we do not include google-services.json in our project, but still change the annotation in `FormsNetworkInteractorComponent` to `@FirebaseStorage` instead of `@GenericNetworkStorage` the app will crash with `ClassNotFoundException`. We need to gracefully handle this inside the code somehow (and also document the annotation about this issue).

**Comment by chinmoy12c** [comment]
Hey @098suraj, are you planning to complete this?

**Comment by 098suraj** [comment]
Hey @chinmoy12c, I have made a few changes kindly look into it.
I have used `getIdentifier` to get the  google app id as we can't check if google services json is available at runtime.
So, I check for the  google app id  and if it is present, we can use `FirebaseStorageHandler` or `GenericNetworkStorageHandler` as a default case.

File: odk/extension/src/main/java/io/samagra/odk/collect/extension/components/FormsNetworkInteractorComponent.kt [hunk 4]
```diff
@@ -14,7 +14,7 @@ import javax.inject.Singleton
 @Component(modules = [FormsNetworkInteractorModule::class])
 interface FormsNetworkInteractorComponent {
 
-    @GenericNetworkStorage
+    @FirebaseStorage
```
Comment (chinmoy12c) [inline odk/extension/src/main/java/io/samagra/odk/collect/extension/components/FormsNetworkInteractorComponent.kt]: Please leave this at `@GenericNetworkStorage`.

**Review by chinmoy12c — CHANGES_REQUESTED** [review]

File: odk/extension/src/main/java/io/samagra/odk/collect/extension/components/FormsNetworkInteractorComponent.kt (hunk 4 above)
Comment (098suraj) [reply odk/extension/src/main/java/io/samagra/odk/collect/extension/components/FormsNetworkInteractorComponent.kt]: I have done the changes, @chinmoy12c kindly check it.

**Pull request merged**